src/controller/CHIPDeviceController.cpp

```diff
@@ -803,6 +812,37 @@ CHIP_ERROR DeviceCommissioner::EstablishPASEConnection(NodeId remoteDeviceId, Re
             ExitNow(err = CHIP_ERROR_INVALID_ARGUMENT);
         }
     }
+#endif
+#if CHIP_DEVICE_CONFIG_ENABLE_WIFIPAF
+    if (params.GetPeerAddress().GetTransportType() == Transport::Type::kWiFiPAF)
+    {
+        if (DeviceLayer::ConnectivityMgr().GetWiFiPAF()->GetWiFiPAFState() != Transport::WiFiPAFBase::State::kConnected)
+        {
+            ChipLogProgress(Controller, "WiFi-PAF: Subscribing the NAN-USD devices");
+            static constexpr useconds_t kWiFiStartCheckTimeUsec = WIFI_START_CHECK_TIME_USEC;
+            static constexpr uint8_t kWiFiStartCheckAttempts    = WIFI_START_CHECK_ATTEMPTS;
+
+            DeviceLayer::ConnectivityMgrImpl().StartWiFiManagement();
+            {
+                for (int cnt = 0; cnt < kWiFiStartCheckAttempts; cnt++)
+                {
+                    if (DeviceLayer::ConnectivityMgrImpl().IsWiFiManagementStarted())
+                    {
+                        break;
+                    }
+                    usleep(kWiFiStartCheckTimeUsec);
+                }
+            }
+            if (!DeviceLayer::ConnectivityMgrImpl().IsWiFiManagementStarted())
+            {
+                ChipLogError(NotSpecified, "Wi-Fi Management taking too long to start - device configuration will be reset.");
+            }
```

Comment:
This doesn't feel like the kind of code that belongs in this method, especially with the retries and busy waiting.

Reply:
I will move it to the initialization function